x/ccv/provider/keeper/grpc_query.go

```diff
@@ -617,3 +618,59 @@ func (k Keeper) QueryConsumerChain(goCtx context.Context, req *types.QueryConsum
 		PowerShapingParams: &powerParams,
 	}, nil
 }
+
+// QueryConsumerChain returns the genesis time
+// of the consumer chain associated with the provided consumer id
+func (k Keeper) QueryConsumerGenesisTime(goCtx context.Context, req *types.QueryConsumerGenesisTimeRequest) (*types.QueryConsumerGenesisTimeResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	consumerId := req.ConsumerId
+	if err := ccvtypes.ValidateConsumerId(consumerId); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get consumer initialization params. If they aren't found,
+	// it means that there is no consumer for that consumerId.
+	params, err := k.GetConsumerInitializationParameters(ctx, consumerId)
+	if err != nil {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"cannot get consumer genesis time for consumer Id: %s: %s",
+			consumerId, types.ErrUnknownConsumerId,
+		)
+	}
+
+	// Get the consumer clientId. If it isn't found, it means
+	// that the consumer hasn't been launched or has been stopped and deleted.
+	clientID, ok := k.GetConsumerClientId(ctx, consumerId)
+	if !ok {
+		return nil, status.Errorf(
+			codes.InvalidArgument,
+			"cannot get consumer genesis time for consumer Id: %s: consumer hasn't been launched or has been stopped and deleted",
+			consumerId,
+		)
+	}
+
+	// Get the consensus state of the consumer client at the initial height,
+	// for which the timestamps corresponds to the consumer genesis time
+	cs, ok := k.clientKeeper.GetClientConsensusState(
+		ctx,
+		clientID,
+		params.InitialHeight,
```

Comment:
Aren't old consensus states pruned? If yes, this query would stop working after some time.

Reply:
Not sure, needs checking.

However, we can make it so that the query runs for chains that haven't yet started (passed spawntime).

Reply:
Those consumers will not have a client and thus no genesis time.

Reply:
@insumity The reason for this query is starting the consumer chain -- creating the genesis file with a correct timestamp that will not result in expired consumer clients on the provider. The consensus state will not be pruned at least until a new client update is received by the provider (otherwise the client update will fail). By this time, the consumer chain is already running so the query is no longer needed.

Reply:
I thought he was saying that after starting a chain (making it run, with relaying enabled) the initial height consensu state would get pruned. Thus, state at block 1 would not be queryable from `GetClientConsensusState`.

Reply:
Consensus states are eventually pruned, but it takes a while. Otherwise client updates would not work. So for sure, the initial consensus state will remain on the provider for a while. The important part is that it will remain there for sure until either the consumer chain launches or the consumer light client on the provider expires.

Reply:
I see. Thanks for the explanation.